### PR TITLE
Set up association

### PR DESCRIPTION
**Pull request by Hassaanjbaig-code**
## in this pull request

> [`Demo Video`](https://img.youtube.com/vi/vkkgrhD6aXQ/0.jpg)](https://www.youtube.com/watch?v=vkkgrhD6aXQ)

> Create an `app.rb` file that will serve as your console app entry-point.  Also, complete the [rest of the requirements](https://github.com/microverseinc/curriculum-ruby/blob/d75bb1fd092519fe76f12753a0a8684f44d31110/oop/school_library_ui.md?plain=1#LL27C3-L33C44)

> In your `main.rb` define the entry point, this will be a method called main invoked at the end of your file.Also, complete the [rest of the requirements](https://github.com/microverseinc/curriculum-ruby/blob/d75bb1fd092519fe76f12753a0a8684f44d31110/oop/school_library_ui.md?plain=1#LL34C3-L38C32)

### CODE REVIEW

**Review by FrederickMih — COMMENTED** [review]
## Hi :raising_hand:  @Hassaanjbaig-code , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

**Linters are failing on this PR.**

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  end
+  puts 'Select a person to see their rentals by there id'
+  rental_id = gets.chomp.to_i
+  @rentals.each do |rental|
+    if rental.person.id == rental_id
+      puts "Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}, Person: #{rental.person.name}"
+    end
+    display
+  end
+end
```
Comment (muneebulrehman) [inline app.rb]: While listing the rentals, it only displays one rental for student even though I did create multiple and no rentals for teacher. Kindly fix this issue. I have recorded [this video](https://www.loom.com/share/4351345e7baf4807af273d62bfab9994) where i demonstrated what exactly is the issue.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @Hassaanjbaig-code , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @Hassaanjbaig-code, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made changes as required :heavy_check_mark: 
 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**